fhir-server/src/main/java/com/ibm/fhir/server/util/FHIRRestHelper.java

```diff
@@ -2444,13 +2447,32 @@ private Bundle createSearchBundle(List<Resource> resources, FHIRSearchContext se
                                             .id(UUID.randomUUID().toString())
                                             .total(totalCount);
 
+        // Calculate how many resources are 'match' mode
+        int pageSize = searchContext.getPageSize();
+        int offset = (searchContext.getPageNumber() - 1) * pageSize;
+        int matchResourceCount = pageSize;
+        if (totalCount.getValue() < offset + pageSize) {
+            matchResourceCount = totalCount.getValue() - offset;
+        }
+
+        // Check if too many included resources
+        if (resources.size() > matchResourceCount + SearchConstants.MAX_PAGE_SIZE) {
```

Comment:
the resources list can't ever include any OperationOutcome right?  i assume that gets added in to the response bundle at some later point...

Reply:
Correct. At this point, the resources list will only contain resources returned from the query. For issue #1273 we might add OperationOutcome resources, but that could be done after this check.